coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -241,3 +235,65 @@ func (bp *BatchProverTask) recoverActiveAttempts(ctx *gin.Context, batchTask *or
 		log.Error("failed to recover batch active attempts", "hash", batchTask.Hash, "error", err)
 	}
 }
+
+func (bp *BatchProverTask) getBatchTaskDetail(ctx context.Context, dbBatch *orm.Batch, dbChunks []*orm.Chunk, chunkInfos []*message.ChunkInfo, chunkProofs []*message.ChunkProof) (*message.BatchTaskDetail, error) {
+	taskDetail := &message.BatchTaskDetail{
+		ChunkInfos:  chunkInfos,
+		ChunkProofs: chunkProofs,
+	}
+
+	if encoding.CodecVersion(dbBatch.CodecVersion) != encoding.CodecV3 && encoding.CodecVersion(dbBatch.CodecVersion) != encoding.CodecV4 {
+		return taskDetail, nil
+	}
+
+	chunks := make([]*encoding.Chunk, len(dbChunks))
+	for i, c := range dbChunks {
+		blocks, getErr := bp.blockOrm.GetL2BlocksInRange(ctx, c.StartBlockNumber, c.EndBlockNumber)
+		if getErr != nil {
+			return nil, fmt.Errorf("failed to get blocks in range for batch %d, chunk %d, (start: %d, end: %d): %w", dbBatch.Index, c.Index, c.StartBlockNumber, c.EndBlockNumber, getErr)
+		}
+		chunks[i] = &encoding.Chunk{Blocks: blocks}
+	}
+
+	dbParentBatch, getErr := bp.batchOrm.GetBatchByIndex(ctx, dbBatch.Index-1)
+	if getErr != nil {
+		return nil, fmt.Errorf("failed to get parent batch header for batch %d: %w", dbBatch.Index, getErr)
+	}
+
+	batch := &encoding.Batch{
+		Index:                      dbBatch.Index,
+		TotalL1MessagePoppedBefore: dbChunks[0].TotalL1MessagesPoppedBefore,
+		ParentBatchHash:            common.HexToHash(dbParentBatch.Hash),
+		Chunks:                     chunks,
+	}
+
+	if encoding.CodecVersion(dbBatch.CodecVersion) == encoding.CodecV3 {
+		daBatch, createErr := codecv3.NewDABatch(batch)
+		if createErr != nil {
+			return nil, fmt.Errorf("failed to create DA batch (v3) for batch %d: %w", dbBatch.Index, createErr)
+		}
+		taskDetail.BlobBytes = daBatch.BlobBytes()
+
+		batchHeader, decodeErr := codecv3.NewDABatchFromBytes(dbBatch.BatchHeader)
```

Comment:
Why don't get `dbBatch.BatchHeader from `orm.Batch`?

Reply:
This is got from `orm.Batch.BatchHeader`, and the type fetched from db is `byte array`, so change it to `codecv3.Batch` by `codecv3.NewDABatchFromBytes`.

Reply:
changed to use db value in [bb4d3eb](https://github.com/scroll-tech/scroll/pull/1485/commits/bb4d3eb87ba75615a8a4126e8b83a99cd4d7523a).